documentation/writing_tests.md

```diff
@@ -0,0 +1,1473 @@
+
+- [Mocha](https://www.npmjs.com/package/mocha), which offers a well known and stable test framework for both NodeJS and the browser.
+- [Chai](https://www.npmjs.com/package/chai), a well known assertion library for Node and the browser.
+- [Rollup](https://www.npmjs.com/package/rollup), to bundle JavaScript for different environments, which helps us write in TypeScript and build for Node and the browser.
+- [Karma](https://www.npmjs.com/package/karma), which allows us to run our tests in multiple browsers.
+
+We're also making use of internal tools:
+
+- The [Recorder](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/test-utils/recorder) is a tool that helps us run our live tests against static recordings obtained from a previous successful run, which aims to ensure that our code hasn't changed while benefitting from not having to reach out to live services every time we want to run our tests.
+
+We'll now explore how we're using the mentioned tools.
+
+### Mocha
+
+Mocha is a well known JavaScript test framework that runs in NodeJS and in the browser. If you're new to Mocha, we recommend you to dive into [Mocha's documentation](https://mochajs.org/).
+
+In this section, we'll explore how the Azure SDK for JS is using mocha, what versions we're using, how we're configuring it, and other recommendations on how to use this library.
+
+#### Mocha in our dependencies
+
+If you want to add `mocha` as a dev dependency to a new project, please use [Rush](https://rushjs.io/) since we're most likely using the same version of Mocha in more projects.once you are in the folder of that project, you can run the following command:
+
+```
+rush add --dev -p mocha
+```
+
+We're using the following Mocha plugins and related dependencies:
+
+- [`@types/mocha`](https://www.npmjs.com/package/@types/mocha) provides the type definitions for mocha.
+- [`mocha-junit-reporter`](https://www.npmjs.com/package/mocha-junit-reporter), which produces JUnit-style XML test results.
+- [`nyc`](https://www.npmjs.com/package/nyc) is [Istanbul](https://istanbul.js.org/)'s command line interface.
+- [`esm`](https://www.npmjs.com/package/esm), a popular ECMAScript module loader.
+- [`source-map-support`](https://www.npmjs.com/package/source-map-support), which provides [source map](https://github.com/mozilla/source-map) support for stack traces in node via the [V8 stack trace API](https://v8.dev/docs/stack-trace-api).
+
+A full `rush add` command that includes `mocha` and all of the previous dependencies follows:
+
+```
+rush add --dev -p mocha @types/mocha mocha-junit-reporter nyc esm source-map-support
```

Comment:
Does rush support the addition of multiple dependencies at once?

Reply:
Yes, sir!

Reply:
You're right! I will split these